internal/messages/state.go

```diff
@@ -27,14 +27,8 @@ var _ Message = (*State)(nil)
 // Type implements Message.
 func (s *State) Type() Type { return TypeState }
 
-// Invalidates implements Message.
-func (s *State) Invalidates(m Message) bool {
-	other, ok := m.(*State)
-	if !ok {
-		return false
-	}
-	return s.NodeName == other.NodeName && s.Time > other.Time
```

Comment:
Checking for time here wasn't necessary. I didn't realize that the queue is guaranteed to be ordered, so you can assume that you're always checking whether a message added more recently invalidates a previous message.

This is how memberlist handles its Invalidates internally; [it only ever checks for name](https://github.com/hashicorp/memberlist/blob/432d7e1bae280bb8924c648e84fed49a6617989a/broadcast.go#L32), and not for any kind of [lamport time](https://github.com/hashicorp/memberlist/blob/432d7e1bae280bb8924c648e84fed49a6617989a/net.go#L153).

Since we don't need it (and NamedBroadcast doesn't support that kind of time checking), it's removed in favour of just checking against names.